src/modules/todolist.js

```diff
@@ -0,0 +1,159 @@
+  divContainer2.className = 'w-auto';
+
+  const elCompleted = document.createElement('input');
+  const elDescription = document.createElement('label');
+  const elDescriptionEditor = document.createElement('input');
+  const img = document.createElement('img');
+
+  elCompleted.type = 'checkbox';
+  elDescriptionEditor.type = 'text';
+
+  elCompleted.checked = todo.completed;
+  elDescription.textContent = todo.description;
+  elDescriptionEditor.value = todo.description;
+  elDescriptionEditor.className = 'hidden borderless';
+  elDescription.className = `${todo.completed ? 'cancelled' : ''}`;
+  img.src = moreImage;
+
+  const removeBtn = document.createElement('img');
+  removeBtn.src = binImage;
+  removeBtn.className = 'hidden';
+
+  const startEdit = () => {
+    divTodo.classList.toggle('bg-brown');
+    elDescription.classList.toggle('hidden');
+    elDescriptionEditor.classList.toggle('hidden');
+    img.classList.toggle('hidden');
+    removeBtn.classList.toggle('hidden');
+    elDescriptionEditor.focus();
+  };
+
+  img.addEventListener('click', (event) => {
+    event.stopImmediatePropagation();
+    startEdit();
+  });
+  elDescription.addEventListener('click', (event) => {
+    event.stopImmediatePropagation();
+    startEdit();
+  });
+
+  removeBtn.addEventListener('click', (event) => {
+    event.stopImmediatePropagation();
+    removeTodo(todo);
+  });
+  elDescriptionEditor.addEventListener('click', (event) => {
+    event.stopImmediatePropagation();
+  });
+  elDescriptionEditor.addEventListener('keypress', (event) => {
+    if (event.key === 'Esc') {
+      stopEdits();
+    } else if (event.key === 'Enter') {
+      divTodo.classList.toggle('bg-brown');
+      elDescription.textContent = elDescriptionEditor.value;
+      todo.description = elDescriptionEditor.value;
+      updateTodo(todo);
+      elDescription.classList.toggle('hidden');
+      elDescriptionEditor.classList.toggle('hidden');
+      img.classList.toggle('hidden');
+      removeBtn.classList.toggle('hidden');
+    }
+  });
+  elCompleted.addEventListener('change', (event) => {
+    const completed = event.target.checked;
+    todo.completed = completed;
+    updateTodo(todo);
+    elDescription.classList.toggle('cancelled');
+  });
+  divContainer1.appendChild(elCompleted);
+  divContainer1.appendChild(elDescription);
+  divContainer1.appendChild(elDescriptionEditor);
+  divContainer2.appendChild(img);
+  divContainer2.appendChild(removeBtn);
+  divTodo.appendChild(divContainer1);
+  divTodo.appendChild(divContainer2);
+  todoListElement.appendChild(divTodo);
+}
+
+function renderTodoList() {
+  const storedvalue = localStorage.getItem(todoStorageName);
+  todoListElement.textContent = '';
+  todoList = [];
+  if (storedvalue !== null) {
+    todoList = JSON.parse(storedvalue);
+  }
+  todoList.forEach((todo) => {
+    addTodo(todo);
+  });
+}
+
+export {
+    clearCompleted, stopEdits, addTodo, renderTodoList, saveData,
+    getTodoList,
+};
```

Comment:
Your todolist.js file is too long. For the best practices you should separate all functions into another js files such as 
 addTodo, clearCompleted etc.,.